cluster/gce/gci/health-monitor.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is for master and node instance health monitoring, which is
+# packed in kube-manifest tarball. It is executed through a systemd service
+# in cluster/gce/gci/<master/node>.yaml. The env variables come from an env
+# file provided by the systemd service.
+
+sleep_seconds=10
+max_seconds=10
+
+echo "Start kubernetes health monitoring"
+
+KUBE_ENV="/home/kubernetes/kube-env"
+if [[ ! -e "${KUBE_ENV}" ]]; then
+  echo "The ${KUBE_ENV} file does not exist!! Terminate cluster initialization."
+  exit 1
+fi
+
+source "${KUBE_ENV}"
+# We simply kill the process when there is a failure. Another upstart job will automatically
+# restart the process.
+while [ 1 ]; do
+  if ! timeout 10 docker version > /dev/null; then
```

Comment:
should the docker & kubelet checks be coupled this way (where they run serially)? In the debian image we run them in parallel.